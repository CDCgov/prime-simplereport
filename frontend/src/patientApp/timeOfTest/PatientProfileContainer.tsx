import { useState } from "react";
import { Redirect } from "react-router-dom";
import { connect, useSelector } from "react-redux";

import Button from "../../app/commonComponents/Button/Button";
import PatientTimeOfTestContainer from "../PatientTimeOfTestContainer";

import PatientProfile from "./PatientProfile";

const PatientProfileContainer = () => {
  const [nextPage, setNextPage] = useState(false);
  const [editPage, setEditPage] = useState(false);
  const plid = useSelector((state: any) => state.plid);
  const patient = useSelector((state) => (state as any).patient as any);

  if (editPage) {
    return (
      <Redirect
        push
        to={{
          pathname: "/patient-info-edit",
          search: `?plid=${plid}`,
        }}
      />
    );
  }

  if (nextPage) {
    return (
      <Redirect
        to={{
          pathname: "/patient-info-symptoms",
          search: `?plid=${plid}`,
        }}
      />
    );
  }

  const buttonGroup = (
    <>
      <div className="margin-top-3">
        <Button
          id="patient-confirm-and-continue"
          label={"Confirm and continue"}
          onClick={() => {
            setNextPage(true);
          }}
        />
      </div>
      <Button
        id="edit-patient-profile-button"
        className="margin-top-1"
        variant="outline"
        label={"Edit information"}
        onClick={() => {
          setEditPage(true);
        }}
      />
    </>
  );

  return (
    <PatientTimeOfTestContainer currentPage={"profile"}>
      <PatientProfile patient={patient} />
      {buttonGroup}
    </PatientTimeOfTestContainer>
  );
};

export default connect()(PatientProfileContainer);
