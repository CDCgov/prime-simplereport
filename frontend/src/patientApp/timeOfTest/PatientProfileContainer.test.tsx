import { MockedProvider } from "@apollo/client/testing";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import PatientProfileContainer from "./PatientProfileContainer";

const mockStore = configureStore([]);

jest.mock("react-router-dom", () => {
  const original = jest.requireActual("react-router-dom");
  return {
    ...original,
    useNavigate: jest.fn(),
  };
});

describe("PatientProfileContainer", () => {
  it("renders", () => {
    const store = mockStore({
      patient: {
        residentCongregateSetting: true,
        employedInHealthcare: true,
        birthDate: "",
        street: "8 Pine Ct",
      },
      plid: "definitely not null I promise",
    });
    render(
      <Provider store={store}>
        <MockedProvider mocks={[]} addTypename={false}>
          <PatientProfileContainer />
        </MockedProvider>
      </Provider>
    );

    expect(
      screen.getAllByText("Profile information", { exact: false })[0]
    ).toBeInTheDocument();
  });
});
