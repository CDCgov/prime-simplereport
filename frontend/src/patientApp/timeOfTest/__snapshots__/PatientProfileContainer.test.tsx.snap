// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientProfileContainer snapshot 1`] = `
<main
  className="patient-app patient-app--form padding-bottom-4"
>
  <div
    className="grid-container maxw-tablet"
  >
    <div
      aria-label="progress"
      className="usa-step-indicator  usa-step-indicator--counters-sm margin-y-205"
    >
      <ol
        className="usa-step-indicator__segments"
      >
        <li
          aria-current={true}
          className="usa-step-indicator__segment usa-step-indicator__segment--current"
        >
          <span
            className="usa-step-indicator__segment-label"
          >
            Profile information
             
            <span
              className="usa-sr-only"
            >
              completed
            </span>
          </span>
        </li>
        <li
          aria-current={false}
          className="usa-step-indicator__segment"
        >
          <span
            className="usa-step-indicator__segment-label"
          >
            Symptoms and history
             
            <span
              className="usa-sr-only"
            >
              completed
            </span>
          </span>
        </li>
      </ol>
      <div
        className="usa-step-indicator__header"
      >
        <h1
          className="usa-step-indicator__heading"
        >
          <span
            className="usa-step-indicator__heading-counter"
          >
            <span
              className="usa-sr-only"
            >
              Step
            </span>
            <span
              className="usa-step-indicator__current-step margin-right-05"
            >
              1
            </span>
            <span
              className="usa-step-indicator__total-steps"
            >
              of 
              2
            </span>
          </span>
          <span
            className="usa-step-indicator__heading-text"
          >
            Profile information
          </span>
        </h1>
      </div>
    </div>
    <div
      className="prime-formgroup usa-prose"
    >
      <h2
        className="prime-formgroup-heading font-heading-lg"
      >
        General information
      </h2>
      <h3
        className="font-heading-sm"
      >
        Name
      </h3>
      <p>
        undefined
      </p>
      <h3
        className="font-heading-sm"
      >
        Preferred language
      </h3>
      <p>
        Not provided
      </p>
      <h3
        className="font-heading-sm"
      >
        Date of birth
      </h3>
      <p>
        Not provided
      </p>
      <h3
        className="font-heading-sm"
      >
        Phone number
      </h3>
      <p>
        Not provided
      </p>
      <h3
        className="font-heading-sm"
      >
        Address
      </h3>
      <p>
        <span
          className="display-block"
        >
          8 Pine Ct
        </span>
        <span
          className="display-block"
        >
          undefinedundefined
        </span>
      </p>
      <h3
        className="font-heading-sm"
      >
        Email address
      </h3>
      <p
        id="patient-email"
      >
        Not provided
      </p>
      <h2
        className="prime-formgroup-heading font-heading-lg"
      >
        Demographics
      </h2>
      <h3
        className="font-heading-sm"
      >
        Race
      </h3>
      <p>
        Not provided
      </p>
      <h3
        className="font-heading-sm"
      >
        Ethnicity
      </h3>
      <p>
        Not provided
      </p>
      <h3
        className="font-heading-sm"
      >
        Tribal affiliation
      </h3>
      <p>
        Not provided
      </p>
      <h3
        className="font-heading-sm"
      >
        Sex assigned at birth
      </h3>
      <p>
        Not provided
      </p>
      <h2
        className="prime-formgroup-heading font-heading-lg"
      >
        Other
      </h2>
      <h3
        className="font-heading-sm"
      >
        Are you a resident in a congregate living setting?
      </h3>
      <p>
        Yes
      </p>
      <h3
        className="font-heading-sm"
      >
        Are you a health care worker?
      </h3>
      <p>
        Yes
      </p>
    </div>
    <div
      className="margin-top-3"
    >
      <button
        className="usa-button"
        id="patient-confirm-and-continue"
        onClick={[Function]}
        type="button"
      >
        Confirm and continue
      </button>
    </div>
    <button
      className="usa-button usa-button--outline margin-top-1"
      id="edit-patient-profile-button"
      onClick={[Function]}
      type="button"
    >
      Edit information
    </button>
  </div>
</main>
`;
